src/Distributed/distributed_fields.jl

```diff
@@ -1,12 +1,22 @@
 import Oceananigans.Fields: Field, FieldBoundaryBuffers, location
 import Oceananigans.BoundaryConditions: fill_halo_regions!
 
+import Oceananigans.Grids: architecture
+
+using Oceananigans.Fields: validate_field_data, validate_boundary_conditions, validate_indices
+
 function Field((LX, LY, LZ)::Tuple, grid::DistributedGrid, data, old_bcs, indices::Tuple, op, status)
     arch = architecture(grid)
+    validate_field_data((LX, LY, LZ), data, grid, indices)
+    validate_boundary_conditions((LX, LY, LZ), grid, old_bcs)
+    indices = validate_indices(indices, (LX, LY, LZ), grid)
     new_bcs = inject_halo_communication_boundary_conditions(old_bcs, arch.local_rank, arch.connectivity)
-    buffers = FieldBoundaryBuffers(nothing, nothing, nothing, nothing)
+    buffers = FieldBoundaryBuffers(grid, data, new_bcs)
```

Comment:
Do we still need this or are the buffers now stored in the boundary conditions?

Reply:
nono, still in the Fields. I would have to say that having them in the field was very convenient for this PR and it probably will be if we have a common buffer for multiple fields.

Reply:
Do you mean it is more convenient to have buffers in `Field.boundary_buffers` versus something like `Field.boundary_conditions.top.boundary_buffer`? Why is the first more convenient?